File: model/ecuaciones.py
import math
def transporte_logitudinal_arena(p,k,ps,n,hb,a,g):
    # p-Densidad del Mar  1025
    # g aceleracion gravitatoria 9.8 
    # k- Inidice de Rompiente 0.78
    # ps- Densidad Arena 2650
    # n -Coeficiente de porocidad  0.4
    # hb altura 0.29
    # a- angulo de Rompiente 20
    # K- Constante de Komar e Inman 0.77
   

    N=(p * math.sqrt(g))
    D=(16 * math.sqrt(k) )* (ps-p)*(1-n)
   
    
    if D==0:
        return False
    else:
        Q= (N)/ (D) * math.sqrt(math.pow(hb,5))* math.sin(2*a)
        print(Q)
        return Q


def obtniedo_K(P,Q):
    if Q==0:
        return False
    else:
        return P / Q 

File: MoreCoast Web/morecoast/src/ecuaciones.js
import * as mathjs from 'mathjs';

export const calculo = async(den_arena_,den_mar_,indice_,coeficiente_,altura_,angulo_,aceleracion_)=>{
   
    const N = (den_mar_ * mathjs.sqrt(aceleracion_));
    const D=(16 * mathjs.sqrt(indice_))*(den_arena_-den_mar_)*(1-coeficiente_);

    if ( D==0 ){
        console.log("LA division por 0 no permitida")
    }else{
       const Q= ( N / D ) * mathjs.sqrt(mathjs.pow(altura_,5))*mathjs.sin(2*angulo_);
        console.log(Q);
    }
}

